### ISSUE TITLE
Documentation page on using Kueue to run Notebook

### ISSUE THREAD

**Issue by mimowo** [kubernetes-sigs/kueue#3878]
**What would you like to be added**:

A documentation page under https://kueue.sigs.k8s.io/docs/tasks/run/ showing how to use Kueue to run Kubeflow Notebook.

Then we should also update the main README and reference the page from the "Features overview / Integrations" point.

This is a follow up to https://github.com/kubernetes-sigs/kueue/issues/3352

**Why is this needed**:

To showcase Kueue is supporting the running of Notebook. To guide users how to do it.

**Completion requirements**:

This enhancement requires the following artifacts:

- [ ] Docs update

The artifacts should be linked in subsequent comments.

**Comment by mimowo** [comment]
cc @xiongzubiao  @varshaprasad96 @tenzen-y

**Comment by varshaprasad96** [comment]
Since the "kueue. x-k8s. io/ queue-name" label is not being added by the notebook-controller to STS, the notebook integration is not working out of the box.

Just tested it out, and looks like a small update in NB controller is needed.

I'll create a PR with NB controller to add NB CR labels to SS, and once it is done can create a doc PR in here.

**Comment by mimowo** [comment]
Awesome, thanks!

**Comment by kannon92** [comment]
This is still relevant.

**Comment by jrleslie** [comment]
hi @varshaprasad96 @kannon92 - we're interested in the notebooks integration, but trying to understand where it stands. Does this integration work today or does the https://github.com/kubeflow/kubeflow/pull/7674 PR still need to be merged?

**Comment by kannon92** [comment]
> hi [@varshaprasad96](https://github.com/varshaprasad96) [@kannon92](https://github.com/kannon92) - we're interested in the notebooks integration, but trying to understand where it stands. Does this integration work today or does the [kubeflow/kubeflow#7674](https://github.com/kubeflow/kubeflow/pull/7674) PR still need to be merged?

Sad. It seems that still needs to merge. I'll open that up.

Opened up https://github.com/kubeflow/notebooks/pull/788.

**Comment by mimowo** [comment]
@kannon92 @varshaprasad96 what is the status here? Is this working and just lacking documentation (as I initially thought), or there is more to do?

**Comment by kannon92** [comment]
It doesn't work. Kubeflow notebooks v1 doesn't propagate the labels to the statefulset. I proposed https://github.com/kubeflow/notebooks/pull/788 to fix it but the community isn't really trying to fix v1 api for notebooks right now.

We have a fork of this for Openshift where the patch did land. There is work to add a proper Kueue integration for notebooks v2.

**Comment by mimowo** [comment]
Should we talk to mainainers to move forward with https://github.com/kubeflow/notebooks/pull/788 ? I don't see any comments like "not a good idea", it rather looks like it slipped the maintainer's attention. Should we maybe re-open and ping maintainers on slack?

**Comment by kannon92** [comment]
> Should we talk to mainainers to move forward with [kubeflow/notebooks#788](https://github.com/kubeflow/notebooks/pull/788) ? I don't see any comments like "not a good idea", it rather looks like it slipped the maintainer's attention. Should we maybe re-open and ping maintainers on slack?

I've had some conversations via slack and there are concerns merging this integration as its a new feature for v1.

**Comment by mimowo** [comment]
> I've had some conversations via slack and there are concerns merging this integration as its a new feature for v1.

Could we mitigate the concerns by some additional API to drive the copying, or FG? For example the labels are copied are only if the StatefulSet has an annotation "kubeflow/copy-labels".

**Comment by mimowo** [comment]
Who was this discussed with specifically, @andreyvelich ? Surely doing the PR unconditionally as currently may be not best, but I'm sure there should be ways to do this as opt-in, for example by the "kubeflow/copy-labels" annotation.

**Comment by andreyvelich** [comment]
Notebooks WG is focusing on V2 version right now.
We need @christian-heusel @thesuperzapper @andyatmiami to prioritize integration with Kueue as part of the next release.

**Comment by mimowo** [comment]
I see so the main problem is that notebook V1 is no longer maintained, right?

**Comment by andreyvelich** [comment]
Yes, as far as I know, the Notebooks v1 is deprecated and be removed in the future versions.

**Comment by kannon92** [comment]
But notebooks v2 is not yet out either. Its not a great situation imo.

**Comment by kannon92** [comment]
@mimowo are you asking because you have customer demand for notebooks? Or just want to check status of this ticket?

**Comment by andreyvelich** [comment]
I let maintainers to comment here, but previously they target GA right before the KubeCon NA 2026.

**Comment by mimowo** [comment]
Actually, I was asked about the status by our PM @MichalZylinski (might be user request or backlog cleanup). I will let Michal clearify.

**Comment by MichalZylinski** [comment]
Yes, I'm currently cleaning our docs backlog and this one is one of the oldest tasks remaining.

**Comment by mimowo** [comment]
In that case I think it is safe to close - it does not work, and will not work because Notebook v1 is no longer maintained.
